src/Middleware/RequestDecompression/src/RequestDecompressionBody.cs

```diff
@@ -0,0 +1,96 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace Microsoft.AspNetCore.RequestDecompression;
+
+/// <summary>
+/// Stream wrapper that creates a specific decompression stream if necessary.
+/// </summary>
+internal sealed class RequestDecompressionBody : Stream
```

Comment:
Is this class really needed? What would go wrong if we set the `GZipSteam` directly to `Request.Body`?
```
var decompressionStream = decompressionProvider.CreateStream(originalBody);
context.Request.Body = decompressionStream;
```

In case there are something that should be done in a new stream, could it at least use inheritance instead of compositions for the known encodings? The providers returns the custom type which implement a marker interface `IDontNeedWrapper` (Very good name, I know!! 😅) and the middleware only wrap the stream if it needs a wrapper.

```
internal sealed class GZipRequestDecompressionBody : GZipStream, IDontNeedWrapper {}
internal sealed class DeflateRequestDecompressionBody : DeflateStream , IDontNeedWrapper {}
internal sealed class BrotliRequestDecompressionBody : BrotliStream, IDontNeedWrapper {}
```

In this case we save one allocation per request for the known encodings.

Reply:
I've removed the `RequestDecompressionBody` class for now and updated the middleware to set the decompression stream directly to `Request.Body`. 

Just want to clarify, as for wrapping the streams, would the middleware simply check if the stream implements the marker interface (maybe `IUnwrappedDecompressionStream`) and then directly set the steam to `Request.Body` if this is the case? Also, how should the middleware handle wrapping a stream that does need to be wrapped?